src/triage/component/postmodeling/contrast/evaluator.py

```diff
@@ -0,0 +1,543 @@
+        self.model_group_id = model_group_id
+
+        # Retrive model_id metadata from the model_metadata schema
+        model_metadata = pd.read_sql("""
+        WITH
+        individual_model_ids_metadata AS(
+        SELECT m.model_id,
+               m.model_group_id,
+               m.hyperparameters,
+               m.model_hash,
+               m.experiment_hash,
+               m.train_end_time,
+               m.train_matrix_uuid,
+               m.training_label_timespan,
+               mg.model_type,
+               mg.model_config
+            FROM model_metadata.models m
+            JOIN model_metadata.model_groups mg
+            USING (model_group_id)
+            WHERE model_group_id = {}
+            AND model_id = {}
+        ),
+        individual_model_id_matrices AS(
+        SELECT DISTINCT ON (matrix_uuid)
+               model_id,
+               matrix_uuid
+            FROM test_results.predictions
+            WHERE model_id = ANY(
+                SELECT model_id
+                FROM individual_model_ids_metadata
+            )
+        )
+        SELECT metadata.*, test.*
+        FROM individual_model_ids_metadata AS metadata
+        LEFT JOIN individual_model_id_matrices AS test
+        USING(model_id);
+        """.format(self.model_group_id, self.model_id),
+                   con=db_engine)
+
+        # Add metadata attributes to model
+        self.model_type = model_metadata.loc[0, 'model_type']
+        self.hyperparameters = model_metadata.loc[0, 'hyperparameters']
+        self.model_hash = model_metadata.loc[0, 'model_hash']
+        self.train_matrix_uuid = model_metadata.loc[0, 'train_matrix_uuid']
+        self.pred_matrix_uuid = model_metadata.loc[0, 'matrix_uuid']
+        self.train_matrix = None
+        self.pred_matrix = None
+        self.preds = None
+        self.feature_importances = None
+        self.model_metrics = None
+
+    def __str__(self):
+        s = 'Model object for model_id: {}'.format(self.model_id)
+        s += '\n Model Group: {}'.format(self.model_group_id)
+        s += '\n Model type: {}'.format(self.model_type)
+        s += '\n Model hyperparameters: {}'.format(self.hyperparameters)
+        s += '\Matrix hashes (train,test): {}'.format([self.train_matrix_uuid,
+                                                      self.pred_matrix_uuid])
+        return s
+
+    def _load_predictions(self):
+        preds = pd.read_sql("""
+                SELECT model_id,
+                       entity_id,
+                       as_of_date,
+                       score,
+                       label_value,
+                       COALESCE(rank_abs, RANK() OVER(ORDER BY score DESC)) AS rank_abs,
+                       rank_pct,
+                       test_label_timespan
+                FROM test_results.predictions
+                WHERE model_id = {}
+                AND label_value IS NOT NULL
+                """.format(self.model_id),
+                con=db_engine)
```

Comment:
Is this a global variable?

Reply:
Yeah, we were talking about this earlier -- I believe this class must require the specification of a database engine in its constructor -- or something like that -- rather than rely on this global, testing engine.